src/addons/addonapi.cpp

```diff
@@ -90,6 +90,22 @@ void AddonApi::initialize() {
   if (s_constructorCallback) {
     s_constructorCallback(this);
   }
+
+  m_timer.setSingleShot(true);
+}
+
+void AddonApi::setTimedCallback(int interval, const QJSValue& callback) {
+  if (!callback.isCallable()) {
+    logger.debug() << "No callback received";
+    return;
+  }
+
+  // Creates a timer for `interval` msec before executing `callback`
+  m_timer.setInterval(interval);
```

Comment:
Why was this changed to an interval instead of a single shot timer?

And also why was the list dropped? This prevents us from creating more than one timer. If that is the proposed design we'd need to add a check at the top of this function to only allow setting a timer when no other timer is already set... But I'd rather just make it possible to have more than one timer.

Reply:
It's still a singleshot timer, `setInterval(msec)` just sets the timer interval, but doesn't necessarily repeat. But you're right, we don't need this as we can just pass the interval in when we start the timer.

For this particular message, only one timer is necessary, so in an effort to not overengineer and add unused complexity, I have simplified it to one overwriteable timer at a time

Reply:
Duh, with regards to the single shot question...

Hum. Ok, let's go with the YAGNI principle. Still, this needs two things 1. unit tests 2. an error in case a timer is created while there is another timer still ongoing.

Reply:
if a timer is created while one is already running, we will overwrite the original timer